Mail2Bug/WorkItemManagement/MessageToWorkItemMapper.cs

```diff
@@ -50,7 +54,16 @@ public class MessageToWorkItemMapper
             }
 
             // Just a standard conversation - look up the cache based on the conversation ID (or guid)
-            int? workItemId = GetWorkItemIdFromConversationId(message.ConversationGuid, _workItemsCache);
+            int? workItemId = null;
+
+            // If using only the ConversationID, check for it first
+            if (_useConversationGuid)
+            {
+                workItemId = GetWorkItemIdFromConversationId(message.ConversationGuid, _workItemsCache);
+            }
+
+            // So we maintain backwards compatibility, and so we don't lose track of all existing items,
```

Comment:
My comment above about not exposing both options in IIncomingMailMessage would make this kind of handling a bit more problematic.
There are three options to handle that which come to mind (there are probably more, these are just the ones I thought of):
1. Don't support back compat - means there'll be some pain when switching, but after the first couple of days this should subside
  1.1. This could also be mitigated by converting the conversation index field in the items when changing the config
2. Introduce special logic in this function to down-cast the IIncomingEmailMessage to EWSIncomingMailMessage (needs to account for the case that it's not of that type, of course), and implements fall back logic using the EWS object.
3. Introduce the option of doing the mapping while loading the work items - basically, for every item, if the indexing field is larger than the GUID, assume that it's the conversation Index and extract the GUID accordingly - this has the downside of introducing some non-obvious coupling between the email settings and the work item management, but I think it's reasonable and is a bit less hacky compared to down-casting.

I'm fine with either of the three (or another one, if you come up with a different alternative). Can help implementing if needed.